### PR TITLE
Federico oauth

### PR DESCRIPTION
**Pull request by FedericoBaron**
Added OAUTH for logging in and now users can only see their own entries.

### CODE REVIEW

File: env_variables.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+env_variables:
+  OAUTH_CLIENT_ID: '1039088902242-1eanapsqfk1imprmu8fkb7qa1fkegqi3.apps.googleusercontent.com'
+  OAUTH_CLIENT_SECRET: 'LrSSvaG-ppaRs-s3UdybKTRg'
```
Comment (danielahaidar) [inline env_variables.yaml]: After deleting this, it should be ready to merge!

**Review by danielahaidar — APPROVED** [review]

**Review by danielahaidar — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/ListEntriesServlet.java [hunk 2]
```diff
@@ -16,8 +16,11 @@
 
 import com.google.cloud.datastore.Datastore;
 import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.StructuredQuery.CompositeFilter;
```
Comment (jonathantanmy) [inline src/main/java/com/google/sps/servlets/ListEntriesServlet.java:20]: Keep this sorted in alphabetical order.

File: src/main/java/com/google/sps/servlets/ListEntriesServlet.java [hunk 3]
```diff
@@ -38,8 +43,25 @@
...
+    }
+    else {
+        response.sendRedirect("/login");
+    }
+
     Query<Entity> query =
-        Query.newEntityQueryBuilder().setKind("Entry").setOrderBy(OrderBy.desc("timestamp")).build();
+        Query.newEntityQueryBuilder()
+        .setKind("Entry")
+        // .setOrderBy(OrderBy.desc("timestamp"))
```
Comment (jonathantanmy) [inline src/main/java/com/google/sps/servlets/ListEntriesServlet.java]: Should we keep this line?

File: src/main/java/com/google/sps/servlets/LoginCallbackServlet.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+public class LoginCallbackServlet extends AbstractAuthorizationCodeCallbackServlet {
+
+  @Override
+  protected AuthorizationCodeFlow initializeFlow() throws IOException {
+    return OAuthUtils.newFlow();
+  }
+
+  @Override
+  protected String getRedirectUri(HttpServletRequest request) {
+      System.out.println("we do get here");
```
Comment (jonathantanmy) [inline src/main/java/com/google/sps/servlets/LoginCallbackServlet.java]: Remove this line

File: src/main/java/com/google/sps/servlets/LoginCallbackServlet.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+  @Override
+  protected String getUserId(HttpServletRequest request) {
+    return request.getSession().getId();
+  }
+
+  @Override
+  protected void onSuccess(HttpServletRequest request, HttpServletResponse response, Credential credential)
+      throws IOException {
+          System.out.println("we are successful");
```
Comment (jonathantanmy) [inline src/main/java/com/google/sps/servlets/LoginCallbackServlet.java]: This line too

File: src/main/java/com/google/sps/servlets/LoginScreenServlet.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpServletRequest;
+
+@WebServlet("/login_screen")
+public class LoginScreenServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    // response.setContentType("text/html;");
```
Comment (jonathantanmy) [inline src/main/java/com/google/sps/servlets/LoginScreenServlet.java]: Delete all the commented lines

File: src/main/java/com/google/sps/servlets/NewEntryServlet.java [hunk 7]
```diff
@@ -19,6 +19,8 @@
 import com.google.cloud.datastore.Entity;
 import com.google.cloud.datastore.FullEntity;
 import com.google.cloud.datastore.KeyFactory;
+import com.google.sps.OAuthUtils;
```
Comment (jonathantanmy) [inline src/main/java/com/google/sps/servlets/NewEntryServlet.java]: Keep this sorted

File: src/main/java/com/google/sps/servlets/ListEntriesServlet.java (hunk 3 above)
Comment (FedericoBaron) [reply src/main/java/com/google/sps/servlets/ListEntriesServlet.java]: i wanted to set the order by timestamp but wasn't able to figure out how to do it while filtering the entries at the same time so i left it commented out for a future update, but i'll just delete it for the time being.

**Review by danielahaidar — APPROVED** [review]

**Review by jonathantanmy — APPROVED** [review]

**Pull request merged**